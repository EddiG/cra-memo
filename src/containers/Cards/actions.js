// @flow
import * as constants from './constants';
import type { Card } from './types';

export const updateCards = (cards: Array<Card>) => ({
  type: constants.UPDATE_CARDS,
  payload: {
    cards,
  },
});

export const closeCards = () => ({
  type: constants.CLOSE_CARDS,
  payload: {},
});

export const triggerCard = (cardId: String) => ({
  type: constants.TRIGGER_CARD,
  payload: {
    id: cardId,
  },
});

export const triggerCardAsync = (cardId: String) => ({
  type: constants.TRIGGER_CARD_ASYNC,
  payload: {
    id: cardId,
  },
});

export const validateCards = () => ({
  type: constants.VALIDATE_CARDS,
  payload: {},
});
