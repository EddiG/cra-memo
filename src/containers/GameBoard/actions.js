import * as constants from './constants';

export const generateCards = () => ({
  type: constants.GENERATE_CARDS,
  payload: {},
});

export const refreshCards = () => ({
  type: constants.REFRESH_CARDS,
  payload: {},
});
