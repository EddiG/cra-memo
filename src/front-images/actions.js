import * as constants from './constants';

export const preloadImagesRequire = () => ({
  type: constants.PRELOAD_FRONT_IMAGES_REQUIRE,
  payload: {},
});

export const preloadImagesComplete = () => ({
  type: constants.PRELOAD_FRONT_IMAGES_COMPLETE,
  payload: {},
});
