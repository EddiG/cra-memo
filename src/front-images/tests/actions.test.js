import * as actions from '../actions';

describe('preloadImagesRequire', () => {
  it('should return an action PRELOAD_FRONT_IMAGES_REQUIRE', () => {
    expect(actions.preloadImagesRequire()).toMatchSnapshot();
  });
});

describe('preloadImagesComplete', () => {
  it('should return an action PRELOAD_FRONT_IMAGES_COMPLETE', () => {
    expect(actions.preloadImagesComplete()).toMatchSnapshot();
  });
});
