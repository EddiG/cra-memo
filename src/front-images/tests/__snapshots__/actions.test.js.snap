// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preloadImagesComplete should return an action PRELOAD_FRONT_IMAGES_COMPLETE 1`] = `
Object {
  "payload": Object {},
  "type": "PRELOAD_FRONT_IMAGES_COMPLETE",
}
`;

exports[`preloadImagesRequire should return an action PRELOAD_FRONT_IMAGES_REQUIRE 1`] = `
Object {
  "payload": Object {},
  "type": "PRELOAD_FRONT_IMAGES_REQUIRE",
}
`;
